modules/tinymce/src/themes/silver/main/ts/modes/ScrollingContext.ts

```diff
@@ -0,0 +1,54 @@
+import { Bounds, Boxes } from '@ephox/alloy';
+import { Arr, Optional, Strings } from '@ephox/katamari';
+import { Css, PredicateFilter, SugarElement, SugarNode } from '@ephox/sugar';
+
+export interface ScrollingContext {
+  readonly element: SugarElement<HTMLElement>;
+  readonly others: SugarElement<HTMLElement>[];
+}
+
+// TINY-9385: Investigate exactly what makes an element possibly
+// scrollable. This approach does not seem exhaustive. What about:
+// overflow-x, overflow-y etc.
+const nonScrollingOverflows = [ 'visible', 'hidden' ];
+
+export const isScroller = (elem: SugarElement<Node> | any): boolean => {
+  if (SugarNode.isHTMLElement(elem)) {
+    const overflow = Css.get(elem, 'overflow');
+    return Strings.trim(overflow).length > 0 && !Arr.contains(nonScrollingOverflows, overflow);
+  } else {
+    return false;
+  }
+};
+
+// NOTE: Calculating the list of scrolling ancestors each time this function is called might
+// be unnecessary. It will depend on its usage.
+export const detect = (poupSinkElem: SugarElement<HTMLElement>): Optional<ScrollingContext> => {
+  // We don't want to include popuSinkElem in the list of scrollers, so we just use "ancestors"
+  const scrollers: SugarElement<HTMLElement>[] = PredicateFilter.ancestors(poupSinkElem, isScroller) as SugarElement<HTMLElement>[];
+
+  const closestScroller = Arr.head(scrollers);
```

Comment:
Is this variable here to debug stuff could they be combined? Sometimes I wonder if we should have an tap or peak function on Optional to aid debugging.

Reply:
I think it was primarily to provide meaning to what the head of the array is. By making it a variable, it's clear that it's the closest scroller. But I could just do that with the parameter name, and here, I should have given it an `opt` prefix anyway

- [ ] inline `closestScroller`